lib/jekyll-remote-theme/dependency_manager.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module RemoteTheme
+    # Since evaluating a gemspec from third-party themes can lead to arbitrary code
+    # execution, this class safely reads a gemspec file at the root of given theme
+    # and handles any runtime dependencies declared within.
+    class DependencyManager
+      DEPENDENCY_MATCHER = %r!add_(?:runtime_)?dependency!
+      EXTRACT_DEPENDENCY_REGEX = %r!#{DEPENDENCY_MATCHER}(?:[(|\s])["'](.*?[^\\])["']!
+
+      # Returns an array of dependency gem-names.
+      attr_reader :theme_dependencies
+
+      def initialize(theme, whitelist)
+        @theme = theme
+        @whitelist = whitelist
+        @theme_dependencies = []
+      end
+
+      # Reads the gemspec at the given path, line-by-line, checking if the line
+      # contains a dependency declaration.
+      #
+      # Returns nothing, but stores the match in an array which can be accessed by
+      # calling +:theme_dependencies+
+      def extract_dependencies
+        return if @gemspec.nil? || !@theme_dependencies.empty?
+
+        File.read(@gemspec).each_line do |line|
+          next unless line =~ DEPENDENCY_MATCHER
+          line.match(EXTRACT_DEPENDENCY_REGEX)
+          @theme_dependencies << Regexp.last_match(1)
```

Comment:
Can these three lines be simplified to `@theme_dependencies = File.read(@gemspec).scan(/$\s*#{EXTRACT_DEPENDENCY_REGEX}/)`?

Reply:
The regex doesn't seem correct to me.. but `.scan(/\s*#{EXTRACT_DEPENDENCY_REGEX}/).flatten` should work..